File: beginner/java/06_count_vowels/Solution.java
import java.util.Arrays;

/**
 * Challenge 06: Count Vowels
 * TODO: Implement the countVowels method
 */
public class Solution {
    
    /**
     * Count the number of vowels in a string
     * 
     * @param text A string to analyze
     * @return The number of vowels (a, e, i, o, u, A, E, I, O, U)
     */
    public static int countVowels(String text) {
        int restult = 0;
        Character[] vowels = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};

        for (int i = 0; i < text.length(); i++) {
            char c = text.charAt(i);
            if (Arrays.asList(vowels).contains(c)) {
                restult++;
            }
        }
        return restult;
    }
    
    public static void main(String[] args) {
        // Test your method
        System.out.println("countVowels(\"hello\") = " + countVowels("hello"));
        System.out.println("countVowels(\"AEIOU\") = " + countVowels("AEIOU"));
        System.out.println("countVowels(\"xyz\") = " + countVowels("xyz"));
        System.out.println("countVowels(\"Programming\") = " + countVowels("Programming"));
    }
}

File: beginner/python/06_count_vowels/solution.py
"""
Challenge 06: Count Vowels
TODO: Implement the count_vowels function
"""

def count_vowels(text: str) -> int:
    """
    Count the number of vowels in a string.
    
    Args:
        text: A string to analyze
    
    Returns:
        int: The number of vowels (a, e, i, o, u, A, E, I, O, U)
    """
    return count_vowels_helper(text) 

def count_vowels_helper(text):
    vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
    count = 0
    for char in text:
        if char in vowels:
            count += 1
    return count


if __name__ == "__main__":
    # Test your function
    print(f"count_vowels('hello') = {count_vowels('hello')}")
    print(f"count_vowels('AEIOU') = {count_vowels('AEIOU')}")
    print(f"count_vowels('xyz') = {count_vowels('xyz')}")
    print(f"count_vowels('Programming') = {count_vowels('Programming')}")
